### PR TITLE
Copy Ben's Webpack 4 to 5 upgrade

### PR DESCRIPTION
**Pull request by kienstra**
* Copies [Ben's Webpack 4 to 5 upgrade](https://github.com/getflywheel/local-addon-instant-reload/pull/21)
* Before, `yarn` failed with:

```sh
gtetflywheel/local-addon-image-optimizer/node_modules/loader-runner/lib/LoaderRunner.js:133
		if(isError) throw e;
		            ^

Error: error:0308010C:digital envelope routines::unsupported

```
* This is because NPM packages were outdated.

### Steps to test
1. Start Local with its `master` branch
2. In this repo: `yarn build && npm pack`
3. You'll see a `.tar.gz` file in this repo's root
4. Go to a site with at least 1 large image on it
5. Tools > Image Optimizer > Scan for Images > Optimize Images:

Screenshot 2023-09-26 at 4 04 23 PM

6. Expected: 
optimized-images

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,51 +2,21 @@
...
+    executor: node/default
     steps:
       - checkout
-      - *restore_yarn_cache
-      - *install_yarn_deps
-      - *save_yarn_cache
-      - *build
-      - *test
+      - node/install:
+          install-yarn: true
```
Comment (kienstra) [inline .circleci/config.yml:14]: This config works the same as before.

Except it'll get the Node version from `.nvmrc`, and we won't have to change it if we bump the Node version.

**Comment by kienstra** [comment]
This is needed before we can do https://github.com/getflywheel/local-addon-image-optimizer/pull/78

File: webpack.config.js [hunk 2]
```diff
@@ -53,11 +51,6 @@ module.exports = {
 			},
 		],
 	},
-	node: {
-		fs: 'empty',
-		child_process: 'empty',
-		__dirname: false,
-	},
```
Comment (kienstra) [inline webpack.config.js:60]: [Thanks to Ben](https://github.com/getflywheel/local-addon-instant-reload/pull/21) for this.

**Review by nickcernis — APPROVED** [review]

**Comment by kienstra** [comment]
Thanks a lot for reviewing this, @nickcernis!

**Pull request merged**